slippy/tile.go

```diff
@@ -2,77 +2,123 @@ package slippy
 
 import (
 	"errors"
-	"fmt"
+	"math"
+	"strconv"
 
 	"github.com/go-spatial/geom"
 )
 
+var (
+	ErrNilBounds = errors.New("slippy: Bounds cannot be nil")
+)
+
 // MaxZoom is the lowest zoom (furthest in)
 const MaxZoom = 22
 
-// NewTile returns a Tile of Z,X,Y passed in
-func NewTile(z, x, y uint) *Tile {
-	return &Tile{
+// Zoom represents a zoom level; this usually goes from 0 to 22
+type Zoom uint
+
+func (z Zoom) N() float64 { return math.Exp2(float64(z)) }
+func (z Zoom) TileSize() Tile {
+	n := int(z.N())
+	return Tile{
 		Z: z,
-		X: x,
-		Y: y,
+		X: n,
+		Y: n,
 	}
 }
 
 // Tile describes a slippy tile.
 type Tile struct {
 	// zoom
-	Z uint
+	Z Zoom
 	// column
-	X uint
+	X int
 	// row
-	Y uint
+	Y int
```

Comment:
Because there is a case where a negative number makes sense (-1 is the only one) which is to get the pixels outside of the map. But now that i think about it we don't really use it. So, i should change it back.